wallet/wallet-node-client/src/rpc_client/mod.rs

```diff
@@ -0,0 +1,170 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod client_impl;
+
+use std::sync::atomic::AtomicU64;
+
+use common::{
+    chain::{Block, GenBlock},
+    primitives::{BlockHeight, Id},
+};
+use serialization::hex::{HexDecode, HexEncode};
+use ureq::serde::de::Error;
+
+#[derive(thiserror::Error, Debug)]
+pub enum NodeRpcError {
+    #[error("Initialization error: {0}")]
+    InitializationError(Box<NodeRpcError>),
+    #[error("Request error: {0}")]
+    RequestError(Box<ureq::Error>),
+    #[error("Response error: {0}")]
+    JsonResponseError(#[from] serde_json::Error),
+    #[error("Response json interpretation error: {0}")]
+    ResponseJsonInterpretationError(std::io::Error),
+    #[error("Decoding error: {0}")]
+    DecodingError(#[from] serialization::hex::HexError),
+    #[error("Json response type error")]
+    JsonResponseTypeError,
+}
+
+pub struct NodeRpcClient {
+    host: String,
+}
+
+static REQ_ID: AtomicU64 = AtomicU64::new(0);
+
+impl NodeRpcClient {
+    pub fn new(remote_socket_address: String) -> Result<Self, NodeRpcError> {
+        let host = format!("http://{remote_socket_address}");
+        let client = Self { host };
+
+        // Attempt to communicate with the node to make sure connection works
+        client
+            .get_best_block_id()
+            .map_err(|e| NodeRpcError::InitializationError(Box::new(e)))?;
+
+        Ok(client)
+    }
+
+    fn make_request(&self, req_data: serde_json::Value) -> Result<serde_json::Value, NodeRpcError> {
+        let resp = ureq::post(self.host.as_str())
+            .set("User-Agent", "AuthServiceProxy/0.1")
+            .set("Content-type", "application/json")
+            .send_json(req_data)
+            .map_err(|e| NodeRpcError::RequestError(Box::new(e)))?;
+        let json_response = resp
+            .into_json::<serde_json::Value>()
+            .map_err(NodeRpcError::ResponseJsonInterpretationError)?;
+
+        Self::error_if_json_error(&json_response)?;
+
+        Ok(json_response)
+    }
+
+    /// If the json response contains the error field, return an error.
+    fn error_if_json_error(json_response: &serde_json::Value) -> Result<(), NodeRpcError> {
+        if let Some(err) = json_response.get("error") {
+            return Err(NodeRpcError::JsonResponseError(serde_json::Error::custom(
+                err.to_string(),
+            )));
+        }
+        Ok(())
+    }
+
+    fn get_block(&self, block_id: Id<Block>) -> Result<Option<Block>, NodeRpcError> {
+        let block_id_hex = block_id.hex_encode();
+
+        let req_data = ureq::json!({
+            "method": "chainstate_get_block",
+            "jsonrpc": "2.0",
+            "id": REQ_ID.fetch_add(1, std::sync::atomic::Ordering::SeqCst),
+            "params": ureq::json!({
+                "id": block_id_hex,
+            }),
+        });
```

Comment:
I think we should not keep RPC in sync manually. Why not use generated `jsonrpsee` client instead?

Reply:
I agree... but besides that I don't know how to do that, there's a problem with whether we're gonna even keep jsonrpcsee due to lack of authentication. Until that's resolved, I don't really care how this works.